src/plugins/release-monitor/index.js

```diff
@@ -0,0 +1,292 @@
+ * @param {string} description - description for the status
+ * @returns {Promise} Resolves when the status is created on the PR
+ * @private
+ */
+function createStatusOnPR({ context, state, sha, description }) {
+    return context.github.repos.createStatus(
+        context.repo({
+            sha,
+            state,
+            target_url: "",
+            description,
+            context: "Release monitor"
+        })
+    );
+}
+
+/**
+ * Create success status on the PR
+ * @param {Object} context - probot context object
+ * @param {string} sha - sha for the commit
+ * @returns {Promise} Resolves when the status is created on the PR
+ * @private
+ */
+function createSuccessPRStatus({ context, sha, description = "No patch release is pending" }) {
+    return createStatusOnPR({
+        context,
+        state: "success",
+        description,
+        sha
+    });
+}
+
+/**
+ * Create pending status on the PR
+ * @param {Object} context - probot context object
+ * @param {string} sha - sha for the commit
+ * @returns {Promise} Resolves when the status is created on the PR
+ * @private
+ */
+function createPendingPRStatus({ context, sha }) {
+    return createStatusOnPR({
+        context,
+        state: "pending",
+        description: "A patch release is pending",
+        sha
+    });
+}
+
+/**
+ * Get all the commits for a PR
+ * @param {Object} context - probot context object
+ * @param {string} number - pr number
+ * @returns {Promise} Resolves when the all PR objects are retrieved
+ * @private
+ */
+function getAllCommitForPR(context, number) {
+    return context.github.pullRequests.getCommits(
+        context.repo({
+            number
+        })
+    );
+}
+
+/**
+ * Updates the status on all the provided prs
+ * @param {Object} context - probot context object
+ * @param {Array<Object>} prs - Collection of prs
+ * @param {boolean} isSuccess - to create a success status
+ * @returns {Promise} Resolves when the status is created on the PRs
+ * @private
+ */
+async function createStatusOnPRs({ context, prs, isSuccess }) {
+    const statusFunc = isSuccess ? createSuccessPRStatus : createPendingPRStatus;
+
+    await Promise.all(prs.map(async pr => {
+        const allCommits = await getAllCommitForPR(context, pr.number);
+
+        await statusFunc({
+            context,
+            sha: pluckLatestCommitSha(allCommits)
+        });
+    }));
+}
+
+/**
+ * Get all the commits for a PR
+ * @param {Object} context - probot context object
+ * @param {boolean} isSuccess - to create a success status
+ * @returns {Promise} Resolves when the status is created on the PR
+ * @private
+ */
+async function createStatusOnAllPRs({ context, isSuccess }) {
+    const { data: allOpenPrs } = await getAllOpenPRs(context);
+
+    return createStatusOnPRs({
+        context,
+        prs: allOpenPrs,
+        isSuccess
+    });
+}
+
+/**
+ * Returns collection of all the PRs which do not have fix or docs tag on it
```

Comment:
Nit: Maybe this should say "which are not semver-patch" (semver-patch = more than fix/docs)